doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+// Returns: a matrix
+select(database:"foo").range(start:-4h)
+```
+
+#### Clear
+`clear` will remove series from a matrix that have no points so they are not returned in the response.
+
+```javascript
+select(database:"foo", where:{'metric'="cpu"})
+  .range(start:-1h)
+  .clear()
+```
+
+#### Casting types
+The value types in a matrix can be cast using the type casting functions. When converting from string to int64, float64 or others they will be parsed. Errors can either be ignored or returned.
+
+##### Float
+`float` will convert a single value or all values in a matrix to `float64` type values.
+
+```javascript
+// Accepts: matrix, int64, uint64, bool, string
+// Arguments:
+//   fixed - an integer specifying how many decimal places to round to
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of float64 values or other value converted to a float64
+```
+
+##### Int
+`int` will convert a single value or all values in a matrix to `int64` type values.
+
+```javascript
+// Accepts: matrix, float64, uint64, bool, string
+// Arguments:
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of int64 values or other value converted to a int64
+```
+
+##### UInt
+`uint` will convert a single value or all values in a matrix to `uint64` type values.
+
+```javascript
+// Accepts: matrix, float64, int64, bool, string
+// Arguments:
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of uint64 values or other value converted to a uint64
+```
+
+##### String
+`string` will convert a single value or all values in a matrix to `string` type values.
+
+```javascript
+// Accepts: matrix, float64, int64, uint64, bool
+// Arguments:
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of string values or other value converted to a string
+```
+
+##### Bool
+`bool` will convert a single value or all values in a matrix to `bool` type values.
+
+```javascript
+// Accepts: matrix, float64, int64, uint64, string
+// Arguments:
+//   error - boolean indicating if it should fail on casting error. default is false
+// Returns: matrix of bool values or other value converted to a bool
+```
+
+#### Window
+`window` will group points is each series in a matrix based on the windowing criteria. These windows can then have functions applied to them to produce aggregates or summaries of each.
+
+```javascript
+// Accepts: a matrix
+// Arguments:
+//   every - start a window after this duration. Must be less than or equal to period.
+//   period - windows are for this duration. If only every or period is specified,
+//            the other is set to the same value by default. If the every duration
+//            is less than the period, windows will overlap.
+//   count - number of windows (points) desired. Will adjust bucket durations
```

Comment:
Has this kind of dynamic bucket duration but fixed count window size been requested before?
Kapacitor never received a request for that kind of windowing behavior.

But Kapacitor does support everyCount, periodCount which indicate that the windows should be constructed on the counts of points instead of their times.

Should we add `everyCount` and `periodCount` parameters and remove the `count` parameter?